crates/e2e/tests/e2e/protocol_fee.rs

```diff
@@ -380,8 +370,8 @@ async fn get_quote(
     services.submit_quote(&quote_request).await
 }
 
-fn fee_in_buy_token(order: &Order, quote: &OrderQuote) -> U256 {
-    order.metadata.executed_fee * quote.buy_amount / quote.sell_amount
+fn fee_in_buy_token(order: &Order) -> U256 {
+    order.metadata.executed_fee
```

Comment:
Isn't this now only correct when `order` is a buy order given that fees should now always be computed in the surplus token?
In that case I would add an assert for that.